plugins/inputs/elasticsearch/elasticsearch.go

```diff
@@ -186,25 +193,37 @@ func (e *Elasticsearch) Gather(acc telegraf.Accumulator) error {
 		e.client = client
 	}
 
-	var wg sync.WaitGroup
-	wg.Add(len(e.Servers))
+	if e.ClusterStats {
+		e.serverInfo = make(map[string]serverInfo)
+		for _, serv := range e.Servers {
+			go func(s string, acc telegraf.Accumulator) {
+				info := serverInfo{}
 
-	for _, serv := range e.Servers {
-		go func(s string, acc telegraf.Accumulator) {
-			defer wg.Done()
-			url := e.nodeStatsUrl(s)
-			e.isMaster = false
+				// Gather node ID
+				if err := e.gatherNodeID(&info, s+"/_nodes/_local/name"); err != nil {
```

Comment:
Is this information also in gatherNodeStats?   If so, could we reorder this to go over the node stats first and then go over the cluster stats, skipping one call?

Reply:
It is, but then it should have been called by /_nodes/_local/stats. Cause that seems to be the only way to gather the local nodeID instead of all.
So we could only skip it if we use _local, otherwise this additional call is always needed.
Might be cleaner to keep it just this way?

Reply:
Makes sense, flow is already hard enough to follow so let's keep it how you have it.